### DISCUSSION TITLE
botocore.exceptions.ParamValidationError: Parameter validation failed:

### DISCUSSION THREAD

**Discussion by Cjcascade** [cloud-custodian/cloud-custodian#7406] [General]
I am receiving an error "Unknown parameter in input: "OpsItemType", must be one of: Description, OperationalData, OperationalDataToDelete, Notifications, Priority, RelatedOpsItems, Status, OpsItemId, Title, Category, Severity, ActualStartTime, ActualEndTime, PlannedStartTime, PlannedEndTime."

We use CodeBuild to run Cloud Custodian within AWS. 
buildspec.yml file:
version: 0.2
phases:
  install:
    runtime-versions:
      python: 3.7
    commands:
     # - apt-get update -qq
      - pip install --upgrade pip
      - pip install c7n
      - pip install c7n-mailer
      - pip install python-dateutil
      - pip install --upgrade awscli
  build:
    commands:
      - bash -ex build.sh

Part of the CodeBuild Logs: The failure begins at line 420
+ c7n-mailer --config mailer.yml --update-lambda
--
414 | 2021-04-28 18:07:33,375 - custodian-mailer - INFO - Provisioning mailer lambda us-east-1
415 | + custodian run -c policy-shield.yml -s .
416 | 2021-04-28 18:07:34,673: custodian.policy:INFO policy:open-ingress-port-22 resource:security-group region:us-east-1 count:0 time:0.35
417 | 2021-04-28 18:07:36,033: custodian.policy:INFO policy:iam-user-access-keys-older-than-90days resource:iam-user region:us-east-1 count:0 time:1.36
418 | 2021-04-28 18:07:36,045: custodian.policy:INFO policy:iam-user-service-access-keys-older-than-365days resource:iam-user region:us-east-1 count:0 time:0.01
419 | 2021-04-28 18:07:36,047: custodian.policy:INFO policy:iam-passoword-older-than-90days resource:iam-user region:us-east-1 count:2 time:0.00
420 | 2021-04-28 18:07:36,240: custodian.output:ERROR Error while executing policy
421 | Traceback (most recent call last):
422 | File "/root/.pyenv/versions/3.7.10/lib/python3.7/site-packages/c7n/policy.py", line 317, in run
423 | results = a.process(resources)
424 | File "/root/.pyenv/versions/3.7.10/lib/python3.7/site-packages/c7n/resources/ssm.py", line 438, in process
425 | resources = self.update_items(client, items, dict(item_template), resources)
426 | File "/root/.pyenv/versions/3.7.10/lib/python3.7/site-packages/c7n/resources/ssm.py", line 540, in update_items
427 | client.update_ops_item(**i)
428 | File "/root/.pyenv/versions/3.7.10/lib/python3.7/site-packages/botocore/client.py", line 357, in _api_call
429 | return self._make_api_call(operation_name, kwargs)
430 | File "/root/.pyenv/versions/3.7.10/lib/python3.7/site-packages/botocore/client.py", line 649, in _make_api_call
431 | api_params, operation_model, context=request_context)
432 | File "/root/.pyenv/versions/3.7.10/lib/python3.7/site-packages/botocore/client.py", line 697, in _convert_to_request_dict
433 | api_params, operation_model)
434 | File "/root/.pyenv/versions/3.7.10/lib/python3.7/site-packages/botocore/validate.py", line 293, in serialize_to_request
435 | raise ParamValidationError(report=report.generate_report())
436 | botocore.exceptions.ParamValidationError: Parameter validation failed:
437 | Unknown parameter in input: "OpsItemType", must be one of: Description, OperationalData, OperationalDataToDelete, Notifications, Priority, RelatedOpsItems, Status, OpsItemId, Title, Category, Severity, ActualStartTime, ActualEndTime, PlannedStartTime, PlannedEndTime
438 | 2021-04-28 18:07:36,241: custodian.commands:ERROR Error while executing policy iam-passoword-older-than-90days, continuing
439 | Traceback (most recent call last):
440 | File "/root/.pyenv/versions/3.7.10/lib/python3.7/site-packages/c7n/commands.py", line 271, in run
441 | policy()
442 | File "/root/.pyenv/versions/3.7.10/lib/python3.7/site-packages/c7n/policy.py", line 1184, in __call__
443 | resources = mode.run()
444 | File "/root/.pyenv/versions/3.7.10/lib/python3.7/site-packages/c7n/policy.py", line 317, in run
445 | results = a.process(resources)
446 | File "/root/.pyenv/versions/3.7.10/lib/python3.7/site-packages/c7n/resources/ssm.py", line 438, in process
447 | resources = self.update_items(client, items, dict(item_template), resources)
448 | File "/root/.pyenv/versions/3.7.10/lib/python3.7/site-packages/c7n/resources/ssm.py", line 540, in update_items
449 | client.update_ops_item(**i)
450 | File "/root/.pyenv/versions/3.7.10/lib/python3.7/site-packages/botocore/client.py", line 357, in _api_call
451 | return self._make_api_call(operation_name, kwargs)
452 | File "/root/.pyenv/versions/3.7.10/lib/python3.7/site-packages/botocore/client.py", line 649, in _make_api_call
453 | api_params, operation_model, context=request_context)
454 | File "/root/.pyenv/versions/3.7.10/lib/python3.7/site-packages/botocore/client.py", line 697, in _convert_to_request_dict
455 | api_params, operation_model)
456 | File "/root/.pyenv/versions/3.7.10/lib/python3.7/site-packages/botocore/validate.py", line 293, in serialize_to_request
457 | raise ParamValidationError(report=report.generate_report())
458 | botocore.exceptions.ParamValidationError: Parameter validation failed:
459 | Unknown parameter in input: "OpsItemType", must be one of: Description, OperationalData, OperationalDataToDelete, Notifications, Priority, RelatedOpsItems, Status, OpsItemId, Title, Category, Severity, ActualStartTime, ActualEndTime, PlannedStartTime, PlannedEndTime
460 | 2021-04-28 18:07:36,242: custodian.commands:ERROR The following policies had errors while executing
461 | - iam-passoword-older-than-90days
462 |  
463 | [Container] 2021/04/28 18:07:36 Command did not exit successfully bash -ex build.sh exit status 2
464 | [Container] 2021/04/28 18:07:36 Phase complete: BUILD State: FAILED
465 | [Container] 2021/04/28 18:07:36 Phase context status code: COMMAND_EXECUTION_ERROR Message: Error while executing command: bash -ex build.sh. Reason: exit status 2
466 | [Container] 2021/04/28 18:07:36 Entering phase POST_BUILD
467 | [Container] 2021/04/28 18:07:36 Phase complete: POST_BUILD State: SUCCEEDED
468 | [Container] 2021/04/28 18:07:36 Phase context status code:  Message:

The policy:
- name: iam-passoword-older-than-90days
    description: |
        Retrieve all IAM user accounts whom have a password that are older than 90 days
    resource: iam-user
    filters:
      - type: credential
        key: password_last_changed
        value_type: age
        value: 90
        op: greater-than
      - type: credential
        key: user
        op: regex
        # Match all values that do not start with svc_ (case insenstive)
        value: '^(?i)(?!svc_).+$'
    actions:
      - type: post-item
        title: IAM user password greater than 90 days
      - type: notify
        template: default
        priority_header: '2'
        subject: WARN-CLOUD-AWS - Detected an IAM user password greater than 90 days in account ###ENVIRONMENT_NAME###
        to:
          - ###OWNER###
          - ###HBS_SECURITY_CONTACT###
        transport:
          type: sqs
          queue: https://sqs.us-east-1.amazonaws.com/###ACCOUNT_NUMBER###/aws-coreinfra-custodian-queue 

The issue seems to be within the "actions" portion of the policy.
If I comment out that portion, it executes successfully.

We have this policy running in 5 accounts - it runs successful within 2 and fails within 3. All 5 accounts are running the same code from the same repository.

**Comment by kapilt** [comment]
thats super odd, re the differential behavior by account, possibly this is an api change in ssm ops manager thats not evenly rolled out.

[update] that looks like it might be differential between a post and an update of an item.

**Comment by Cjcascade** [comment]
I've been able to temporarily resolve it by commenting out the "post-item" action on that particular policy. The policy is now completing successfully and the email notifications are being sent via ses if the policy is in violation.
